File: backend-javascript/src/subscriptions/billing.controller.ts
import { Body, Controller, Post } from '@nestjs/common';
import { SubscriptionsService } from './subscriptions.service';

@Controller('billing')
export class BillingController {
  constructor(private readonly subscriptionsService: SubscriptionsService) {}

  @Post('confirm')
  confirm(@Body() body: { token: string }) {
    return this.subscriptionsService.confirmPaymentByToken(body.token);
  }

  @Post('stripe/confirm')
  confirmStripe(@Body() body: { sessionId: string }) {
    return this.subscriptionsService.confirmStripeSession(body.sessionId);
  }
}

File: backend-java/src/main/java/com/neria/manager/subscriptions/BillingController.java
package com.neria.manager.subscriptions;

import java.util.Map;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/billing")
public class BillingController {
  private final SubscriptionsService subscriptionsService;

  public BillingController(SubscriptionsService subscriptionsService) {
    this.subscriptionsService = subscriptionsService;
  }

  @PostMapping("/confirm")
  public Map<String, Object> confirm(@RequestBody Map<String, String> body) {
    return subscriptionsService.confirmPaymentByToken(body.getOrDefault("token", ""));
  }

  @PostMapping("/stripe/confirm")
  public Map<String, Object> confirmStripe(@RequestBody Map<String, String> body) {
    return subscriptionsService.confirmStripeSession(body.getOrDefault("sessionId", ""));
  }
}
